### ISSUE TITLE
`bun install` required after every `bun add` & `bun remove`

### ISSUE THREAD

**Issue by ryandotelliott** [oven-sh/bun#9136]
### What version of Bun is running?

1.0.29+a146856d1

### What platform is your computer?

Darwin 23.1.0 arm64 arm

### What steps can reproduce the bug?

I created a new [Turborepo](https://turbo.build/repo/docs/getting-started/create-new) project using `bun create next-app` and setting up Turbo in addition, I had this issue since very early into the project. After running any `bun add` or `bun remove` in either of my workspace directories, I am required to run `bun install` after, within the same directory, otherwise I get an error.

### What is the expected behavior?

The package should be added/removed without issue.

### What do you see instead?

I received the error:
```
FileNotFound
Bun could not find a file, and the code that produces this error is missing a better error.

----- bun meta -----
Bun v1.0.29 (a146856d) macOS Silicon 23.1.0
AddCommand: dotenv 
Elapsed: 3ms | User: 6ms | Sys: 5ms
RSS: 20.22MB | Peak: 20.22MB | Commit: 34.62MB | Faults: 0
----- bun meta -----
```

<img width="661" alt="image" src="https://github.com/oven-sh/bun/assets/12452699/cd3d6664-da01-4d52-9dd5-8a2385a83d7c">

### Additional information

I have two workspaces:
```
"workspaces": [
  "apps/*",
  "packages/*"
]
```

My repo is setup as follows:
<img width="286" alt="image" src="https://github.com/oven-sh/bun/assets/12452699/62224964-370f-4378-9c5a-296efd839f1a">

**Comment by Lucas-Eduardo-Goncalves** [comment]
I received a similar error:
![image](https://github.com/oven-sh/bun/assets/92813658/4afd0da8-4bbd-4f09-8788-7f53950afd17)

This is a github repository:
https://github.com/Lucas-Eduardo-Goncalves/Arkyn

this is a `./package.json`:
![image](https://github.com/oven-sh/bun/assets/92813658/2ac45fd5-3368-412b-af68-33103954935c)

this is a `./packages/types/package.json`: 
![image](https://github.com/oven-sh/bun/assets/92813658/4318702f-db94-4878-b77e-6778bcffab15)

this is a `./packages/components/package.json`: 
![image](https://github.com/oven-sh/bun/assets/92813658/7823fac2-4739-46b0-b778-3d232183e53b)

when executing `bun i` I receive as console
![image](https://github.com/oven-sh/bun/assets/92813658/498b6441-5ffe-4b72-8437-ea439ef17f86)

But when trying to install or remove any dependency on the `@arkyn/components` package I get the error
![image](https://github.com/oven-sh/bun/assets/92813658/d82e3822-1ee9-4e41-9518-6f10596bcc3b)

The only way to get around this error is to remove dependencies on internal packages when installing an NPM package.

**Comment by zachsents** [comment]
Same issue here on 1.0.31

**Comment by oof2win2** [comment]
This bug is only present when using the `workspace:*` version specifier. When using a different specifier (such as `0.0.1`), this issue is resolved.

**Comment by webbertakken** [comment]
Might be related to changing cwd context. 

I just had the same error running `bun install --cwd someNonExistentFolder`

![image](https://github.com/oven-sh/bun/assets/20756439/37d87769-a3bc-44fa-8744-6c243c492d29)

Creating the `someNonExistentFolder`, or in my case `prod2`, fixed the issue.

![image](https://github.com/oven-sh/bun/assets/20756439/2d22f556-8b89-4cb1-a5c4-70391240a0ce)

**Comment by dylan-conway** [comment]
This was fixed in bun `v1.1.9` with #11177
